### PR TITLE
Add `json()` method for HTTP Request and Response classes.

### PR DESCRIPTION
**Pull request by rbdixon**
#### Description

Simple `.json()` method modeled after what Requests does.

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Review by mhils — COMMENTED** [review]
This looks useful, thanks! 🍰

File: mitmproxy/http.py [hunk 1]
```diff
@@ -494,6 +496,20 @@ def encode(self, encoding: str) -> None:
...
 
+    def json(self, **kwargs: Any) -> Optional[dict]:
+        """
+        Returns the JSON encoded content of the response, if any. If the body is empty
+        then `None` will be returned. `**kwargs` are optional arguments that will be
+        passed to `json.loads()`.
+
+        *Raises:*
+
+          - `json.decoder.JSONDecodeError`, if content can not be decoded as JSON.
```
[the remark this answers is not in the document]
Comment (rbdixon) [reply mitmproxy/http.py]: Done.

File: mitmproxy/http.py [hunk 2]
```diff
@@ -494,6 +496,20 @@ def encode(self, encoding: str) -> None:
...
+        """
+        if self.content:
+            return json.loads(self.content, **kwargs)
+        return None
```
[the remark this answers is not in the document]
Comment (rbdixon) [reply mitmproxy/http.py]: Done. I had to add a check for the type of `Flow().get_content()` because `None` is a possible return value.

**Comment by mhils** [comment]
Fantastic, thanks again! 🍰

**Pull request merged**